### PR TITLE
TCVP-1410 add open telemetry to oracle data api rebase

### PR DESCRIPTION
**Pull request by karahanb**
# Description

This PR includes the following proposed change(s):

- [TCVP-1410](https://justice.gov.bc.ca/jira/browse/TCVP-1410)
- Added configuration to trace oracle-data-api through Open Telemetry with Jaeger exporter.
- Added configuration for enabling actuator module to get health and monitoring metrics from oracle-data-api.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [x] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [x] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/oracle-data-api/src/main/resources/application.yml [hunk 1]
```diff
@@ -2,6 +2,17 @@ server:
...
+  application:  
+    name: oracle-data-api
+    
+  sleuth:
+    otel:
+      config:
+        trace-id-ratio-based: 1.0
+      exporter:
+        otlp:
+          endpoint: ${OTEL_EXPORTER_OTLP_ENDPOINT:http://localhost:4317}
```
Comment (pbolduc) [inline src/backend/oracle-data-api/src/main/resources/application.yml:14]: we should be able to send directly to jaeger.

**Review by pbolduc — COMMENTED** [review]
I dont think we require sleuth.

File: .docker/docker-compose.jaeger.yml [hunk 2]
```diff
@@ -22,8 +22,26 @@ services:
...
-
-  ticket-search:
+      
+  oracle-data-api:
     environment:
-      OTEL_EXPORTER_JAEGER_ENDPOINT: http://jaeger:14268/api/traces
-      OTEL_EXPORTER_JAEGER_PROTOCOL: http/thrift.binary
+      OTEL_EXPORTER_OTLP_ENDPOINT: ${OTEL_EXPORTER_OTLP_ENDPOINT:-http://sleuth:4317}
+  
+  sleuth:
```
[the remark this answers is not in the document]
Comment (karahanb) [reply .docker/docker-compose.jaeger.yml:30]: I really tried couple of different ways but could not make it directly work with the jaeger instance to send traces. That's why I had to add a separate collector. This separate collector may also enable us to collect different metrics and telemetry data and export to other desired observability tools in the correct format in the future. Would it cause any issues if we use a separate collector like this?

**Comment by KDNeufeld** [comment]
I get an error starting this with docker-compose:
`docker-compose -f docker-compose.yml -f ./.docker/docker-compose.jaeger.yml up -d --build`
```
21:29:06.924 [main] ERROR org.springframework.boot.SpringApplication - Application run failed
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 1024
	at org.yaml.snakeyaml.reader.StreamReader.peek(StreamReader.java:136)
	at org.yaml.snakeyaml.scanner.ScannerImpl.scanToNextToken(ScannerImpl.java:1222)
	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchMoreTokens(ScannerImpl.java:308)
	at org.yaml.snakeyaml.scanner.ScannerImpl.checkToken(ScannerImpl.java:248)
	at org.yaml.snakeyaml.parser.ParserImpl$ParseBlockMappingKey.produce(ParserImpl.java:634)
	at org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:165)
```

File: src/backend/oracle-data-api/pom.xml [hunk 3]
```diff
@@ -92,6 +124,36 @@
...
+	   			<!-- Exclude Brave (the default) -->
+	      		<exclusion>
+	        		<groupId>org.springframework.cloud</groupId>
+	        		<artifactId>spring-cloud-sleuth-brave</artifactId>
+	      		</exclusion>
+	    	</exclusions>
+	  	</dependency>
+	  	<dependency>
+	    	<groupId>org.springframework.cloud</groupId>
+	    	<artifactId>spring-cloud-sleuth-otel-autoconfigure</artifactId>
```
[the remark this answers is not in the document]
Comment (karahanb) [reply src/backend/oracle-data-api/pom.xml:146]: Yes, we need this one to configure the app from the application.yml for telemetry tracing. We're still pushing to jaeger directly. I previously named the otel collector container as sleuth in docker but it was not directly related to these dependencies.

**Comment by karahanb** [comment]
> I get an error starting this with docker-compose: `docker-compose -f docker-compose.yml -f ./.docker/docker-compose.jaeger.yml up -d --build`
> 
> ```
> 21:29:06.924 [main] ERROR org.springframework.boot.SpringApplication - Application run failed
> java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 1024
> 	at org.yaml.snakeyaml.reader.StreamReader.peek(StreamReader.java:136)
> 	at org.yaml.snakeyaml.scanner.ScannerImpl.scanToNextToken(ScannerImpl.java:1222)
> 	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchMoreTokens(ScannerImpl.java:308)
> 	at org.yaml.snakeyaml.scanner.ScannerImpl.checkToken(ScannerImpl.java:248)
> 	at org.yaml.snakeyaml.parser.ParserImpl$ParseBlockMappingKey.produce(ParserImpl.java:634)
> 	at org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:165)
> ```

I could not reproduce this error on my local. I tried with the same command and it works fine for me.

**Review by KDNeufeld — APPROVED** [review]
Great work.

**Pull request merged**